### ISSUE TITLE
Inline YouTube videos are blocked on some sites

### ISSUE THREAD

**Issue by nikrolls** [nikrolls/uBlock-Edge#76]
### Describe the issue
On some sites, eg Windows Central, inline YouTube videos are blocked by a `child-src http:` CSP header.

### One or more specific URLs where the issue occurs
http://www.windowscentral.com/askdanwindows-23

### Screenshot in which the issue can be seen

Edge:
![image](https://cloud.githubusercontent.com/assets/2626677/25155286/6dbc5620-24e8-11e7-866c-df0a9f4e520f.png)

Chrome:
![image](https://cloud.githubusercontent.com/assets/2626677/25155309/829d9efa-24e8-11e7-8aac-8f439b0f17d8.png)

### Steps for anyone to reproduce the issue

1. Enable uBlock
2. Browse to http://www.windowscentral.com/askdanwindows-23. Videos are missing.
3. Disable uBlock
4. Refresh the page. Videos are present.

### Your settings
Default.

- OS/version: Microsoft Windows [Version 10.0.16176]
- Browser/version: 40.16176.1000.0
- uBlock Origin version: 1.12.1.0

##### Your filter lists

Default

**Comment by gorhill** [comment]
It's the `|blob:$domain=1337x.to|1channel.biz|ancient-origins.net|androidcentral.com|anime-joy.tv|...` filter in EasyList which causes this. The exception filter `@@|blob:$domain=windowscentral.com` can be used to work around the issue.

Given the site works fine with Chrome/Firefox without the need for the exception filter, I gather this is an issue with Edge?

**Comment by nikrolls** [comment]
@gorhill: Ok, it took me a little while to track down because I was a bit confused as to how Chrome and Firefox were allowing `https:` URLs when the CSP only specified `http:`. But then I found the information regarding this in the CSP3 draft change list: http://www.w3.org/TR/CSP3/#changes-from-level-2

So it looks like Chrome and Firefox are running according to the draft and Edge hasn't released any work on it yet.

A solution is to change the `child-src http:` directive to `child-src http: https:` which works for me on the above URL. I haven't yet tested how it works on sites with existing `child-src` directives it needs to merge with. If merging works as-is and you didn't want to put this tweak in Core then I could add it as a transformation in the Edge build process. If merging needs more work then it would need to be done in Core or fixed in the browser, the latter of which could take time unless they are willing to hot-fix it.

**Comment by nikrolls** [comment]
@scottlow FYI.

**Comment by gorhill** [comment]
> A solution is to change the child-src http: directive to child-src http: https: which works for me on the above URL.

Coincidentally this is something I changed in the commit: <https://github.com/gorhill/uBlock/commit/3ce3ed2a0e1591819340e6fb092fa549ec8c6e12>, part of 1.12.3b1

**Comment by nikrolls** [comment]
Ah, ideal! Thanks @gorhill, nice and easy.

**Comment by AdamDotNet** [comment]
@nikrolls I tried gorhill's `@@|blob:$domain=windowscentral.com` and the youtube iframes load, but then the video ultimately has a playback error. Turning off uBlock to see if the video can play without errors works. Are you seeing that too? Either way, I'm subscribed to this thread to see when a new build pulls in the CSP fix mentioned above, so I'd be wait and see once that's in.

Example: https://www.windowscentral.com/hands-surface-laptop

Image I'm seeing:
![youtube playback error](https://cloud.githubusercontent.com/assets/8978001/25667333/706c2dd4-2fe0-11e7-86d4-40ee361bfcbf.png)

Edge HTML 15 from the Windows 10 Creators Update (stable).

uBlock version 1.12.1.

**Comment by nikrolls** [comment]
Hi @AdamDotNet, do you have large media element blocking turned on?

**Comment by AdamDotNet** [comment]
Settings page under default behavior, right?
"Block media elements larger than  x kB" is **not checked.**

## Filter lists checked:
uBlock Filters
uBlock Filters - Badware risks
uBlock Filters - privacy
uBlock Filters - Unbreak

Adblock protector list
Adblock warning removal list
Anti-Adblock Killer | Reek
Easylist

EasyPrivacy

Malware Domain List
Malware domains
Peter Lowe's Ad and tracking server list

## My filters
windowscentral.com##.mn_mbn_share
@@|blob:$domain=windowscentral.com

**Comment by nikrolls** [comment]
Fixed in https://github.com/nikrolls/uBlock-Edge/releases/tag/1.12.4

**Comment by nikrolls** [comment]
I'll check for your issue @AdamDotNet and create a new ticket shortly if I can reproduce it.

**Comment by cdayo** [comment]
Hi @nikrolls , with 1.12.4 I'm actually getting the error that @AdamDotNet was describing! It happened on any of the windowscentral embedded videos. Now I can't seem to get past the pesky Admiral anti-adblock they have on that site - any thoughts on how to get around this?

**Comment by gorhill** [comment]
> any thoughts on how to get around this?

Force an update of _uBlock filters_.

**Comment by cdayo** [comment]
Ah, that did the trick. Thank you! Now I can happily reproduce @AdamDotNet 's issue! Happening on a different link as below: http://www.windowscentral.com/why-windows-s-and-project-centennial-are-important-continuums-success

![image](https://cloud.githubusercontent.com/assets/28756572/26154909/b16a1c0e-3b54-11e7-8e10-7009aef11cef.png)
